week3/homework/src/todo.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const fs = require('fs');
+const uuid = require('uuid/v4');
+
+const DEFAULT_ENCODING = 'utf8';
+
+class Todo {
+  constructor(filename) {
+    this._filename = filename;
+  }
+
+  async create(description) {
+    const todos = await this.read();
+
+    const todo = {
+      id: uuid(),
+      done: false,
+
+      description
+    };
+
+    todos.push(todo);
+
+    await this._save(todos);
+
+    return todo;
+  }
+
+  read() {
+    return new Promise(resolve => {
+      fs.readFile(this._filename, DEFAULT_ENCODING, (error, data) => {
+        if (error)
+          return resolve([]);
+
+        return resolve(JSON.parse(data));
+      });
+    });
+  }
+
+  async readin(id) {
+    console.log(id)
+    const todos = await this.read();
+    const todo = todos.find(t => t.id == id);
+    return todo;
+
+  }
+
+  async deleteIn() {
```

Comment:
What does `In` in the name `deleteIn()` mean?

Reply:
i really don't know ! ., while am busy with doing my homework i just used a random name , as i always do and i just forget to change it ... i change it for now hope it's better ...

Reply:
That's OK, it can happen! Just try to use meaningful names, so it is easier to understand the purpose.